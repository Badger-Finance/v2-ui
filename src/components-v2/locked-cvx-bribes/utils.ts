import BigNumber from 'bignumber.js';
import { formatWithoutExtraZeros, numberWithCommas, unscale } from '../../mobx/utils/helpers';

export function formatBalance(balance: BigNumber): string {
  return numberWithCommas(formatWithoutExtraZeros(unscale(balance, 18)));
}
