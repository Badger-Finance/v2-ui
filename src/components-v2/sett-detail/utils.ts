import dayjs from 'dayjs';
import utc from 'dayjs/plugin/utc';
import { ChartMode } from '../../mobx/model/setts/sett-charts';
import { DelaySeverity } from '../../mobx/model/setts/sett-rewards';
import { SettDetailMode } from '../../mobx/model/setts/sett-detail';

dayjs.extend(utc);

export const ChartModeTitles = {
	[ChartMode.value]: 'Sett Value',
	[ChartMode.ratio]: 'Token Ratio',
	[ChartMode.accountBalance]: 'Your Holdings',
};

export const SettModeTitles = {
	[SettDetailMode.settInformation]: 'Vault Info',
	[SettDetailMode.userInformation]: 'User Info',
};

export const calculateDelaySeverity = (delay: number): DelaySeverity => {
	if (delay >= 4) {
		return DelaySeverity.high;
	}

	if (delay >= 2) {
		return DelaySeverity.medium;
	}

	return DelaySeverity.none;
};

export const calculateDifferenceInHoursFromCycle = (cycle: Date): number => {
	return Math.abs(dayjs(cycle).diff(dayjs(), 'hours'));
};
