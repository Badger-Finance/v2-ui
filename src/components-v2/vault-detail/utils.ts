import { Vault, VaultData } from '@badger-dao/sdk';
import dayjs from 'dayjs';
import utc from 'dayjs/plugin/utc';
import { ChartMode } from '../../mobx/model/vaults/vault-charts';
import { DelaySeverity } from '../../mobx/model/vaults/vault-rewards';

dayjs.extend(utc);

export const ChartModeTitles: Record<string, string> = {
	[ChartMode.Value]: 'Sett Value',
	[ChartMode.Ratio]: 'Token Ratio',
	[ChartMode.AccountBalance]: 'My Holdings',
	[ChartMode.BoostMultiplier]: 'Badger Boost',
};

export const calculateDelaySeverity = (delay: number): DelaySeverity => {
	if (delay >= 4) {
		return DelaySeverity.high;
	}

	if (delay >= 2) {
		return DelaySeverity.medium;
	}

	return DelaySeverity.none;
};

export const calculateDifferenceInHoursFromCycle = (cycle: Date): number => {
	return Math.abs(dayjs(cycle).diff(dayjs(), 'hours'));
};

export function defaultVaultBalance(vault: Vault): VaultData {
	return {
		address: vault.vaultToken,
		name: vault.name,
		symbol: vault.asset,
		pricePerFullShare: vault.pricePerFullShare,
		balance: 0,
		value: 0,
		earnedBalance: 0,
		earnedValue: 0,
		depositedBalance: 0,
		withdrawnBalance: 0,
		tokens: [],
		earnedTokens: [],
	};
}

export interface Balance {
	balance: number;
}

export function hasBalance(input?: Balance): boolean {
	if (!input) {
		return false;
	}
	return input.balance > 0;
}
