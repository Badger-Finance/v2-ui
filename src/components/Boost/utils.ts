import { makeStyles } from '@material-ui/core/styles';
import { ClassNameMap } from '@material-ui/core/styles/withStyles';
import React from 'react';

type ComparisonConfig = {
  toCompareValue: number;
  toBeComparedValue: number;
  greaterCaseColor: string;
  lessCaseColor: string;
  defaultColor: string;
};

export const getColorFromComparison = ({
  toCompareValue,
  toBeComparedValue,
  greaterCaseColor,
  lessCaseColor,
  defaultColor,
}: ComparisonConfig): string => {
  if (toCompareValue > toBeComparedValue) return greaterCaseColor;
  if (toCompareValue < toBeComparedValue) return lessCaseColor;

  return defaultColor;
};

export const useAssetInputStyles = (
  currentValue: string,
  toCompare = 0,
): ((props?: React.CSSProperties) => ClassNameMap<'assetColor'>) => {
  return makeStyles((theme) => {
    const defaultColor = currentValue ? theme.palette.text.primary : theme.palette.text.secondary;
    const fontColor = getColorFromComparison({
      toCompareValue: Number(currentValue),
      toBeComparedValue: toCompare,
      greaterCaseColor: '#74D189',
      lessCaseColor: theme.palette.error.main,
      defaultColor,
    });

    return {
      assetColor: {
        color: fontColor,
      },
    };
  });
};
