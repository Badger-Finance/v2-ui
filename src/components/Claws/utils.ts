import BigNumber from 'bignumber.js';
import { INVALID_REASON } from './index';

export interface ValidationParams {
	amount: string | BigNumber;
	maximum: BigNumber;
	minimum?: BigNumber;
}

export function validateAmountBoundaries({
	amount,
	maximum,
	minimum = new BigNumber('0'),
}: ValidationParams): INVALID_REASON | undefined {
	let error: INVALID_REASON | undefined = undefined;

	const input = new BigNumber(amount);
	const amountExceedsBalance = input.gt(maximum);
	const amountIsLessThanMinimum = input.lt(minimum);

	if (amountIsLessThanMinimum) error = INVALID_REASON.UNDER_MINIMUM;
	if (amountExceedsBalance) error = INVALID_REASON.OVER_MAXIMUM;

	console.log({
		input: input.toString(),
		minimum: minimum.toString(),
		maximum: maximum.toString(),
		amountExceedsBalance,
		amountIsLessThanMinimum,
		error,
	});

	return error;
}
