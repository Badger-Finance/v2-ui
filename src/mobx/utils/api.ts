// api badger functions

// TODO: Update to api.badger.finance on v2 migration
const getApi = () => {
	if (process.env.NODE_ENV === 'production') {
		return 'https://api.sett.vision';
	}
	return 'https://hm77p4zo3l.execute-api.us-west-1.amazonaws.com/staging/';
};
const badgerApi = getApi();

export const getAssetsUnderManagement = (): any => {
	return fetch(`${badgerApi}/protocol/value?tokens=true`).then((response) => response.json());
};

// Geyser Data
export const getFarmData = async (): Promise<any> => {
	return await fetch(`${badgerApi}/protocol/farm`).then((response) => response.json());
};

export const getPpfs = async (): Promise<any> => {
	return await fetch(`${badgerApi}/protocol/ppfs`).then((response) => response.json());
};

export const getAssetPerformances = (setts: Array<any>): Promise<any> => {
	const performanceData = setts.map(async (sett: any) => {
		const assetKey = sett.asset.toLowerCase();
		return {
			...sett,
			asset: assetKey,
			title: sett.title,
			...(await getAssetPerformance(assetKey)),
		};
	});
	return Promise.all(performanceData);
};

const getAssetPerformance = async (asset: any): Promise<any> => {
	return await fetch(`${badgerApi}/protocol/sett/${asset}/performance`).then((response) => response.json());
};

export const getAssetPerformanceCharts = async (setts: Array<any>): Promise<any> => {
	const performanceData = setts.map(async (sett: any) => {
		const assetKey = sett.asset.toLowerCase();
		return {
			asset: assetKey,
			title: sett.title,
			data: await getPerformanceChartData(assetKey),
		};
	});
	return await Promise.all(performanceData);
};

const getPerformanceChartData = async (asset: any): Promise<any> => {
	return await fetch(`${badgerApi}/chart/sett/${asset}/performance?count=350`).then((response) => response.json());
};

export const getSettCharts = async (setts: Array<any>): Promise<any> => {
	const jarData = setts.map(async (sett) => {
		const assetKey = sett.asset.toLowerCase();
		return {
			title: sett.title,
			asset: assetKey,
			data: await getSettChartData(assetKey),
		};
	});
	return await Promise.all(jarData);
};

const getSettChartData = async (asset: any): Promise<any> => {
	return await fetch(`${badgerApi}/chart/sett/${asset}?count=1000`).then((response) => response.json());
};

export const getUserAccount = async (address: string): Promise<any> => {
	return await fetch(`${badgerApi}/protocol/earnings/${address}`).then((response) => response.json());
};

// api coingecko functions

const coingeckoApi = 'https://api.coingecko.com/api/v3';

export const getCoinData = async (coin: string): Promise<any> => {
	return await fetch(`${coingeckoApi}/coins/${coin}`).then((response) => response.json());
};

export const getEthPrice = async (coin: string): Promise<any> => {
	return await fetch(`${coingeckoApi}/coins/${coin}`)
		.then((response) => response.json())
		.then((jsonResponse) => jsonResponse['market_data']['current_price']['usd']);
};

// api util functions

export const formatUsd = (x: number): string => {
	try {
		const valueParts = x.toString().split('.');
		let value = valueParts[0].replace(/\B(?=(\d{3})+(?!\d))/g, ',');
		if (valueParts.length > 1) {
			value += '.' + valueParts[1].substring(0, 2);
		}
		return value;
	} catch (e) {
		console.log(e);
		return x.toString();
	}
};

export const formatWithCommas = (x: number): string => {
	try {
		const valueParts = x.toString().split('.');
		let value = valueParts[0].replace(/\B(?=(\d{3})+(?!\d))/g, ',');
		if (valueParts.length > 1) {
			value += '.' + valueParts[1].substring(0, 5);
		}
		return value;
	} catch (e) {
		console.log(e);
		return x.toString();
	}
};

// api badger functions

export const getBadgerRewards = async (data: string, address: string): Promise<any> => {
	return await fetch(
		`https://fzqm8i0owc.execute-api.us-east-1.amazonaws.com/prod/rewards/1/${data}/${address}`,
	).then((response) => response.json());
};
