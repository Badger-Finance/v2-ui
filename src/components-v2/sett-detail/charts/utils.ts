import dayjs from 'dayjs';
import utc from 'dayjs/plugin/utc';
import { Sett } from '../../../mobx/model/setts/sett';
import { fetchSettChartInformation } from '../../../mobx/utils/apiV2';
import { SettSnapshotGranularity } from '../../../mobx/model/setts/sett-snapshot';
import { SettChartData } from '../../../mobx/model/setts/sett-charts';

dayjs.extend(utc);

export enum SettChartTimeframe {
	'day' = 1,
	'week' = 7,
	'month' = 30,
}

/**
 * Fetches chart information since the provided timeframe until the current date
 * @param sett
 * @param timeframe
 */
export const fetchSettChart = async (sett: Sett, timeframe: SettChartTimeframe): Promise<SettChartData[] | null> => {
	const timeframeDays = 1 * timeframe;
	const isDayTimeFrame = timeframe === SettChartTimeframe.day;

	// if timeframe is just one day then we want the granularity to be hours
	const granularity = isDayTimeFrame ? SettSnapshotGranularity.HOUR : SettSnapshotGranularity.DAY;

	const now = dayjs().utc(); // query until current date
	const from = dayjs(now).subtract(timeframeDays, 'days').utc();

	const fetchedData = await fetchSettChartInformation(sett.vaultToken, from.toDate(), now.toDate(), granularity);

	if (!fetchedData) {
		return null;
	}

	// data needs to be ascending sorted
	return fetchedData
		.sort((a, b) => a.timestamp - b.timestamp)
		.map((d) => ({ ...d, timestamp: new Date(d.timestamp) }));
};
