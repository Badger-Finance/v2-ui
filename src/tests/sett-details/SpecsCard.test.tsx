import SpecsCard from 'components-v2/vault-detail/specs/SpecsCard';
import React from 'react';
import { checkSnapshot } from 'tests/utils/snapshots';

import VaultStore from '../../mobx/stores/VaultStore';
import { SAMPLE_VAULT } from '../utils/samples';

describe('Specs Section', () => {
	it('displays sett information', () => {
		jest.spyOn(VaultStore.prototype, 'getVaultDefinition').mockReturnValue({
			depositToken: { address: SAMPLE_VAULT.underlyingToken, decimals: 18 },
			vaultToken: { address: SAMPLE_VAULT.vaultToken, decimals: 18 },
		});
		checkSnapshot(<SpecsCard vault={SAMPLE_VAULT} />);
	});
});
