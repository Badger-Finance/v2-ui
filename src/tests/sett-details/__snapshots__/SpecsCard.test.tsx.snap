// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Specs Section displays sett information 1`] = `
<div>
  <div
    class="MuiPaper-root-5 MuiCard-root-4 WithStyles(ForwardRef(Card))-root-3 makeStyles-root-1 MuiPaper-elevation1-9 MuiPaper-rounded-6"
  >
    <div
      class="MuiGrid-root-33 makeStyles-specSection-2 MuiGrid-item-35 MuiGrid-grid-xs-true-67"
    >
      <div
        class="MuiGrid-root-33 makeStyles-root-136 MuiGrid-container-34"
      >
        <h6
          class="MuiTypography-root-145 makeStyles-title-142 MuiTypography-h6-155"
        >
          Vault Details
        </h6>
        <hr
          class="MuiDivider-root-177 WithStyles(ForwardRef(Divider))-root-175 WithStyles(ForwardRef(Divider))-root-176"
        />
        <p
          class="MuiTypography-root-145 makeStyles-amount-137 MuiTypography-body1-147"
        >
          <span
            class="MuiSkeleton-root-184 MuiSkeleton-text-185 MuiSkeleton-pulse-188"
            style="width: 209px; height: 37px;"
          />
        </p>
        <p
          class="MuiTypography-root-145 MuiTypography-body2-146"
        >
          Assets Deposited
        </p>
        <div
          class="makeStyles-showMoreContainer-143"
        >
          <div
            class="makeStyles-showMore-144"
          >
            Hide
          </div>
        </div>
        <div
          class="MuiCollapse-container-195 MuiCollapse-entered-196"
          style="min-height: 0px;"
        >
          <div
            class="MuiCollapse-wrapper-198"
          >
            <div
              class="MuiCollapse-wrapperInner-199"
            >
              <div
                class="makeStyles-submetric-139"
              >
                <p
                  class="MuiTypography-root-145 makeStyles-submetricValue-140 MuiTypography-body1-147"
                >
                  1.0082
                </p>
                <span
                  class="MuiTypography-root-145 makeStyles-submetricType-141 MuiTypography-caption-148"
                >
                  tokens per share
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root-33 makeStyles-specSection-2 MuiGrid-item-35 MuiGrid-grid-xs-true-67"
    >
      <div
        class="MuiGrid-root-33 MuiGrid-container-34"
      >
        <p
          class="MuiTypography-root-145 makeStyles-title-200 MuiTypography-body1-147"
        >
          Tokens
        </p>
        <hr
          class="MuiDivider-root-177 WithStyles(ForwardRef(Divider))-root-175 WithStyles(ForwardRef(Divider))-root-201"
        />
        <div
          class="MuiGrid-root-33 MuiGrid-container-34"
        >
          <div
            class="MuiGrid-root-33 makeStyles-tokenSpec-204 MuiGrid-container-34 MuiGrid-justify-xs-space-between-53"
          >
            <div
              class="MuiBox-root-208 MuiBox-root-209"
            >
              <div
                class="makeStyles-tokenImageContainer-206"
              >
                <img
                  alt="Curve.fi renBTC/wBTC/sBTC icon"
                  class="makeStyles-tokenImage-207"
                  src="/assets/icons/curve-renbtc-wbtc-sbtc.png"
                />
              </div>
              <p
                class="MuiTypography-root-145 makeStyles-specName-203 makeStyles-tokenName-205 MuiTypography-body1-147 MuiTypography-colorTextSecondary-171 MuiTypography-displayInline-173"
              >
                Curve.fi renBTC/wBTC/sBTC
              </p>
            </div>
            <h6
              class="MuiTypography-root-145 MuiTypography-subtitle2-157 MuiTypography-displayInline-173"
            >
              2,580
            </h6>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root-33 makeStyles-specSection-2 MuiGrid-item-35 MuiGrid-grid-xs-true-67"
    >
      <div
        class="MuiGrid-root-33 MuiGrid-container-34"
      >
        <div
          class="MuiGrid-root-33 MuiGrid-container-34 MuiGrid-align-items-xs-center-42"
        >
          <div
            class="MuiGrid-root-33 MuiGrid-container-34 MuiGrid-item-35 MuiGrid-grid-xs-7-74"
          >
            <p
              class="MuiTypography-root-145 MuiTypography-body1-147"
            >
              Reward Frequency
            </p>
          </div>
          <div
            class="MuiGrid-root-33 makeStyles-rewardContainer-214 MuiGrid-item-35 MuiGrid-grid-xs-true-67"
          >
            <p
              class="MuiTypography-root-145 makeStyles-reward-213 makeStyles-noneDelayedReward-217 MuiTypography-body1-147 MuiTypography-displayInline-173"
            >
              ~2 Hours
            </p>
          </div>
        </div>
        <hr
          class="MuiDivider-root-177 WithStyles(ForwardRef(Divider))-root-175 WithStyles(ForwardRef(Divider))-root-220"
        />
        <span
          class="MuiTypography-root-145 makeStyles-frequencyDetail-211 MuiTypography-caption-148 MuiTypography-colorTextSecondary-171"
        >
          This Vaultâ€™s rewards are currently being processed.
        </span>
        <a
          class="MuiTypography-root-145 MuiLink-root-221 MuiLink-underlineHover-223 makeStyles-infoLink-212 MuiTypography-colorPrimary-168"
          href="https://badger.wiki/setts#b52fee58e850405abd9701068fd93f37"
          rel="noreferrer"
          target="_blank"
        >
          <div
            class="makeStyles-linkContent-219"
          >
            <span>
              See more
            </span>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root-227 makeStyles-linkIcon-218"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
              />
            </svg>
          </div>
        </a>
      </div>
    </div>
    <div
      class="MuiGrid-root-33 makeStyles-specSection-2 MuiGrid-item-35 MuiGrid-grid-xs-true-67"
    >
      <div>
        <div
          class="makeStyles-titleContainer-253"
        >
          <p
            class="MuiTypography-root-145 MuiTypography-body1-147"
          >
            Fees
          </p>
          <svg
            aria-hidden="true"
            aria-label="see fees descriptions"
            class="MuiSvgIcon-root-227 makeStyles-helpIcon-255 MuiSvgIcon-colorPrimary-228"
            focusable="false"
            title="Click to see full description"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 17h-2v-2h2v2zm2.07-7.75l-.9.92C13.45 12.9 13 13.5 13 15h-2v-.5c0-1.1.45-2.1 1.17-2.83l1.24-1.26c.37-.36.59-.86.59-1.41 0-1.1-.9-2-2-2s-2 .9-2 2H8c0-2.21 1.79-4 4-4s4 1.79 4 4c0 .88-.36 1.68-.93 2.25z"
            />
          </svg>
        </div>
        <hr
          class="MuiDivider-root-177 makeStyles-divider-252"
        />
        <div
          class="MuiGrid-root-33 MuiGrid-container-34"
        >
          <div
            class="MuiGrid-root-33 MuiGrid-container-34 MuiGrid-justify-xs-space-between-53"
          >
            <p
              class="MuiTypography-root-145 makeStyles-specName-267 MuiTypography-body1-147 MuiTypography-colorTextSecondary-171 MuiTypography-displayInline-173"
            >
              Performance Fee
            </p>
            <h6
              class="MuiTypography-root-145 MuiTypography-subtitle2-157 MuiTypography-displayInline-173"
            >
              0.2%
            </h6>
          </div>
          <div
            class="MuiGrid-root-33 MuiGrid-container-34 MuiGrid-justify-xs-space-between-53"
          >
            <p
              class="MuiTypography-root-145 makeStyles-specName-267 MuiTypography-body1-147 MuiTypography-colorTextSecondary-171 MuiTypography-displayInline-173"
            >
              Strategist Performance Fee
            </p>
            <h6
              class="MuiTypography-root-145 MuiTypography-subtitle2-157 MuiTypography-displayInline-173"
            >
              0.1%
            </h6>
          </div>
          <div
            class="MuiGrid-root-33 MuiGrid-container-34 MuiGrid-justify-xs-space-between-53"
          >
            <p
              class="MuiTypography-root-145 makeStyles-specName-267 MuiTypography-body1-147 MuiTypography-colorTextSecondary-171 MuiTypography-displayInline-173"
            >
              Withdraw Fee
            </p>
            <h6
              class="MuiTypography-root-145 MuiTypography-subtitle2-157 MuiTypography-displayInline-173"
            >
              0.5%
            </h6>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root-33 MuiGrid-item-35 MuiGrid-grid-xs-true-67"
    >
      <div
        class="MuiGrid-root-33 makeStyles-linksContainer-270 MuiGrid-container-34"
      >
        <p
          class="MuiTypography-root-145 MuiTypography-body1-147"
        >
          Links
        </p>
        <hr
          class="MuiDivider-root-177 WithStyles(ForwardRef(Divider))-root-175 WithStyles(ForwardRef(Divider))-root-271"
        />
        <div
          class="MuiGrid-root-33 makeStyles-linkContainer-272 MuiGrid-container-34 MuiGrid-align-items-xs-center-42"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root-227 makeStyles-icon-274 MuiSvgIcon-colorPrimary-228"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3.9 12c0-1.71 1.39-3.1 3.1-3.1h4V7H7c-2.76 0-5 2.24-5 5s2.24 5 5 5h4v-1.9H7c-1.71 0-3.1-1.39-3.1-3.1zM8 13h8v-2H8v2zm9-6h-4v1.9h4c1.71 0 3.1 1.39 3.1 3.1s-1.39 3.1-3.1 3.1h-4V17h4c2.76 0 5-2.24 5-5s-2.24-5-5-5z"
            />
          </svg>
          <a
            class="MuiTypography-root-145 MuiLink-root-221 MuiLink-underlineHover-223 makeStyles-link-273 MuiTypography-colorPrimary-168"
            href="https://docs.badger.com/badger-finance/user-guides/convex-renbtc-wbtc-sbtc"
            rel="noreferrer"
            target="_blank"
          >
            User Guide
          </a>
        </div>
        <div
          class="MuiGrid-root-33 makeStyles-linkContainer-272 MuiGrid-container-34 MuiGrid-align-items-xs-center-42"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root-227 makeStyles-icon-274 MuiSvgIcon-colorPrimary-228"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3.9 12c0-1.71 1.39-3.1 3.1-3.1h4V7H7c-2.76 0-5 2.24-5 5s2.24 5 5 5h4v-1.9H7c-1.71 0-3.1-1.39-3.1-3.1zM8 13h8v-2H8v2zm9-6h-4v1.9h4c1.71 0 3.1 1.39 3.1 3.1s-1.39 3.1-3.1 3.1h-4V17h4c2.76 0 5-2.24 5-5s-2.24-5-5-5z"
            />
          </svg>
          <a
            class="MuiTypography-root-145 MuiLink-root-221 MuiLink-underlineHover-223 makeStyles-link-273 MuiTypography-colorPrimary-168"
            href="https://badger.wiki/strategies#ce634a6ad4b0486288180d775a1552ab"
            rel="noreferrer"
            target="_blank"
          >
            Strategy Diagram
          </a>
        </div>
        <div
          class="MuiGrid-root-33 makeStyles-linkContainer-272 MuiGrid-container-34 MuiGrid-align-items-xs-center-42"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root-227 makeStyles-icon-274 MuiSvgIcon-colorPrimary-228"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3.9 12c0-1.71 1.39-3.1 3.1-3.1h4V7H7c-2.76 0-5 2.24-5 5s2.24 5 5 5h4v-1.9H7c-1.71 0-3.1-1.39-3.1-3.1zM8 13h8v-2H8v2zm9-6h-4v1.9h4c1.71 0 3.1 1.39 3.1 3.1s-1.39 3.1-3.1 3.1h-4V17h4c2.76 0 5-2.24 5-5s-2.24-5-5-5z"
            />
          </svg>
          <a
            class="MuiTypography-root-145 MuiLink-root-221 MuiLink-underlineHover-223 makeStyles-link-273 MuiTypography-colorPrimary-168"
            href="https://curve.fi/sbtc/deposit"
            rel="noreferrer"
            target="_blank"
          >
            Get Deposit Token
          </a>
        </div>
        <div
          class="MuiCollapse-container-195 MuiCollapse-hidden-197"
          style="min-height: 0px;"
        >
          <div
            class="MuiCollapse-wrapper-198"
          >
            <div
              class="MuiCollapse-wrapperInner-199"
            >
              <div
                class="MuiGrid-root-33 makeStyles-linkContainer-272 MuiGrid-container-34 MuiGrid-align-items-xs-center-42"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root-227 makeStyles-icon-274 MuiSvgIcon-colorPrimary-228"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M3.9 12c0-1.71 1.39-3.1 3.1-3.1h4V7H7c-2.76 0-5 2.24-5 5s2.24 5 5 5h4v-1.9H7c-1.71 0-3.1-1.39-3.1-3.1zM8 13h8v-2H8v2zm9-6h-4v1.9h4c1.71 0 3.1 1.39 3.1 3.1s-1.39 3.1-3.1 3.1h-4V17h4c2.76 0 5-2.24 5-5s-2.24-5-5-5z"
                  />
                </svg>
                <a
                  class="MuiTypography-root-145 MuiLink-root-221 MuiLink-underlineHover-223 makeStyles-link-273 MuiTypography-colorPrimary-168"
                  href="https://etherscan.io/address/0xd04c48A53c111300aD41190D63681ed3dAd998eC"
                  rel="noreferrer"
                  target="_blank"
                >
                  Vault Address
                </a>
              </div>
              <div
                class="MuiGrid-root-33 makeStyles-linkContainer-272 MuiGrid-container-34 MuiGrid-align-items-xs-center-42"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root-227 makeStyles-icon-274 MuiSvgIcon-colorPrimary-228"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M3.9 12c0-1.71 1.39-3.1 3.1-3.1h4V7H7c-2.76 0-5 2.24-5 5s2.24 5 5 5h4v-1.9H7c-1.71 0-3.1-1.39-3.1-3.1zM8 13h8v-2H8v2zm9-6h-4v1.9h4c1.71 0 3.1 1.39 3.1 3.1s-1.39 3.1-3.1 3.1h-4V17h4c2.76 0 5-2.24 5-5s-2.24-5-5-5z"
                  />
                </svg>
                <a
                  class="MuiTypography-root-145 MuiLink-root-221 MuiLink-underlineHover-223 makeStyles-link-273 MuiTypography-colorPrimary-168"
                  href="https://etherscan.io/address/0x03f7724180AA6b939894B5Ca4314783B0b36b329"
                  rel="noreferrer"
                  target="_blank"
                >
                  Strategy Address
                </a>
              </div>
              <div
                class="MuiGrid-root-33 makeStyles-linkContainer-272 MuiGrid-container-34 MuiGrid-align-items-xs-center-42"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root-227 makeStyles-icon-274 MuiSvgIcon-colorPrimary-228"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M3.9 12c0-1.71 1.39-3.1 3.1-3.1h4V7H7c-2.76 0-5 2.24-5 5s2.24 5 5 5h4v-1.9H7c-1.71 0-3.1-1.39-3.1-3.1zM8 13h8v-2H8v2zm9-6h-4v1.9h4c1.71 0 3.1 1.39 3.1 3.1s-1.39 3.1-3.1 3.1h-4V17h4c2.76 0 5-2.24 5-5s-2.24-5-5-5z"
                  />
                </svg>
                <a
                  class="MuiTypography-root-145 MuiLink-root-221 MuiLink-underlineHover-223 makeStyles-link-273 MuiTypography-colorPrimary-168"
                  href="https://etherscan.io/address/0x075b1bb99792c9E1041bA13afEf80C91a1e70fB3"
                  rel="noreferrer"
                  target="_blank"
                >
                  Underlying Token Address
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          class="makeStyles-showMoreContainer-268"
        >
          <div
            class="makeStyles-showMore-269"
          >
            Show More
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
